dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java

```diff
@@ -273,6 +273,34 @@ private Object getClassFromCDI(Class<?> clazz) {
         return serviceObject;
     }
 
+    /**
+     * @param clazz
+     * @return
+     */
+    @FFDCIgnore(value = { Exception.class })
+    private ManagedObject<?> getClassFromServiceObject(Class<?> clazz, Object serviceObject) {
+
+        if (! clazz.equals(serviceObject.getClass())) {
```

Comment:
This block will never be executed, right?  From line 239: `Class<?> clazz = serviceObject.getClass();`

So, `clazz` will will also be equal to `serviceObject.getClass()`. 

Is the intent to determine if the serviceObject is already a Weld proxy?  If so, it might be better to just check for that directly - like maybe make a public `isWeldProxy(Object o)` method in CDI code that we could call from here, and you could call from the `CDIManagedObjectFactoryImpl` above?

Reply:
This is just a better safe than sorry thing. This method requires serviceObject to be instance of clazz. So why not explicitly check for it? 

The practical difference is that if this is here we'll return null if someone else calls with a mismatched clazz and serviceObject instead of a java error when AbstractManagedObjectFactory<T>.createManagedObject(T instance... is called with something that's not T.